android/sdl_android/src/main/java/com/smartdevicelink/transport/utl/SdlDeviceListener.java

```diff
@@ -174,9 +185,67 @@ public void handleMessage(@NonNull Message msg) {
                     break;
 
                 case com.smartdevicelink.transport.SdlRouterService.MESSAGE_READ:
+                    onPacketRead((SdlPacket) msg.obj);
                     break;
             }
         }
+
+        public void sendStartService() {
+            SdlDeviceListener sdlListener = this.provider.get();
+            SdlPacket serviceProbe = SdlPacketFactory.createStartSession(SessionType.RPC, 0x00, (byte)0x01, (byte)0x00, false);
+            serviceProbe.putTag(ControlFrameTags.RPC.StartService.PROTOCOL_VERSION, MAX_PROTOCOL_VERSION.toString());
+            byte[] constructed = serviceProbe.constructPacket();
+            if (sdlListener.bluetoothTransport != null && sdlListener.bluetoothTransport.getState() == MultiplexBluetoothTransport.STATE_CONNECTED) {
+                sdlListener.bluetoothTransport.write(constructed, 0, constructed.length);
+            }
+        }
+
+        public void onPacketRead(SdlPacket packet) {
+            SdlDeviceListener sdlListener = this.provider.get();
+            VehicleType vehicleType = null;
+            if (packet.getFrameInfo() == SdlPacket.FRAME_INFO_START_SERVICE_ACK) {
+                //parse vehicle Type info from connected system
+                if (packet.getVersion() >= 5) {
+                    vehicleType = getVehicleType(packet);
+                }
+                int hashID = BitConverter.intFromByteArray(packet.getPayload(), 0);
```

Comment:
We actually need to add a version check here. The first method of just converting the raw payload works for protocol versions <= 4 as control frame constructed payloads were added in version 5. So for versions >=5 we use the getTage method.

Reply:
@YaroslavLutsenko since we already do a version check for >=5 right above this, you could change it to something like : 

```
                int hashID;
                if (packet.getVersion() >= 5) {
                    vehicleType = getVehicleType(packet);
                    hashID = (Integer) packet.getTag(ControlFrameTags.RPC.StartServiceACK.HASH_ID);
                } else {
                    hashID = BitConverter.intFromByteArray(packet.getPayload(), 0);
                }
```

Reply:
@JulianKast thx, added (0971c14)